src/Interpreters/HashJoin.cpp

```diff
@@ -898,9 +943,10 @@ NO_INLINE IColumn::Filter joinRightColumns(
             }
         }
 
-        auto find_result = key_getter.findKey(map, i, pool);
+        bool row_acceptable = !added_columns.isRowFiltered(i);
+        auto find_result = row_acceptable ? key_getter.findKey(map, i, pool) : decltype(key_getter.findKey(map, i, pool))();
 
-        if (find_result.isFound())
+        if (row_acceptable && find_result.isFound())
```

Comment:
Shouldn't any `FindResult::isFound() == false` by default?

Reply:
Yes it is, checking `row_acceptable` can be removed.